File: ChainResponsibility/python-source/handlers/DivisionZeroHandler.py
from __future__ import annotations

from .BaseHandler import BaseHandler
from .Log import Log


class DivisionZeroHandler(BaseHandler):
    """Звено, которое намеренно падает.

    Нужно, чтобы показать: сбой одного звена не рвёт цепочку. Остальные звенья
    отработают, а про ошибку сообщит заданная функция.
    """

    def process(self, log: Log) -> None:
        _ = 1 / 0

File: JavaSource/src/main/java/ru/akhcheck/patterns/chainresponsibility/handlers/DivisionZeroHandler.java
package ru.akhcheck.patterns.chainresponsibility.handlers;

import java.util.Set;

/**
 * Звено, которое намеренно падает.
 *
 * <p>Нужно, чтобы показать: сбой одного звена не рвёт цепочку. Остальные звенья
 * отработают, а про ошибку сообщит заданная функция.
 */
public class DivisionZeroHandler extends BaseHandler {
    public DivisionZeroHandler(Set<LogLevel> logLevels) {
        super(logLevels);
    }

    @Override
    protected void process(Log log) {
        int zero = 0;
        System.out.println(1 / zero);
    }
}
